test/e2e/testnet/testnet.go

```diff
@@ -340,7 +326,11 @@ func (t *Testnet) Start() error {
 		for i := 0; i < 10; i++ {
 			resp, err := client.Status(context.Background())
 			if err != nil {
-				return fmt.Errorf("node %s status response: %w", node.Name, err)
+				if i == 9 {
+					return fmt.Errorf("node %s status response: %w", node.Name, err)
+				}
+				time.Sleep(time.Second)
+				continue
```

Comment:
There are scenarios where the reverse proxy still needs to be ready. This does not fail now when the status requests return a non `2xx` return code.

Reply:
Is there a way where the reverse proxy can tell us when it's ready? Or a way where we don't need to rely on sleeping

Reply:
@mojtaba-esk

Reply:
When `Knuu` is initialized with proxy enabled, the deploy and check service for proxy is done internally and they are blocking, so once knuu is initialized, the proxy should be ready. There should be no need for sleep and wait.

here: https://github.com/celestiaorg/knuu/blob/4e6dcf097463fd9a58c40d2d7c7c9065ac94d9cc/pkg/traefik/traefik.go#L177-L178

Reply:
The deployment of the proxy is blocking until deplopyed.
But when doing some testing I observed that the proxy takes some time to apply the config for a certain Instance.
When creating a forward for an Instance using the proxy, it makes a custom resource in Kubernetes, which is then picked up by the proxy.
Sometimes, we use the proxy host before the proxy applies this configuration.

Reply:
oh I see, so we probably can add something to the `AddHost` func to check that. or have a callback func if that's needed.

Reply:
would you mind to create an issue for it to keep tracking of it in its own issue?

Reply:
@smuu

Reply:
Created issue in knuu: https://github.com/celestiaorg/knuu/issues/423